Source/Core/arrayRemoveDuplicates.js

```diff
@@ -0,0 +1,89 @@
+/*global define*/
+define(['./defaultValue', './defined', './DeveloperError', './Math'], function(defaultValue, defined, DeveloperError, CesiumMath) {
+    'use strict';
+
+    var removeDuplicatesEpsilon = CesiumMath.EPSILON10;
+
+    /**
+     * Removes adjacent duplicate values in an array of values.
+     *
+     * @param {Object[]} values The array of values.
+     * @param {Object} objectType The object type to use when comparing values. The object type must support the equalsEpsilon(left, right, epsilon) method.
```

Comment:
It also needs to support `clone`.  While this approach is fine, another option would be to have them pass in two functions (clone and equalsEpsilon) instead of the object.